backend/test/model_test/institution_test.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+
+from ..test_base import TestBase
+
+from models.institution import Institution
+from models.request_institution_parent import RequestInstitutionParent
+from permissions import DEFAULT_ADMIN_PERMISSIONS
+
+from .. import mocks
+
+
+class InstitutionTest(TestBase):
+    """Test the institution model."""
+
+    @classmethod
+    def setUp(cls):
+        """Provide base for tests."""
+        cls.test = cls.testbed.Testbed()
+        cls.test.activate()
+        cls.policy = cls.datastore.PseudoRandomHRConsistencyPolicy(
+            probability=1)
+        cls.test.init_datastore_v3_stub(consistency_policy=cls.policy)
+        cls.test.init_memcache_stub()
+
+        """Init models"""
+        cls.admin = mocks.create_user()
+        cls.user = mocks.create_user()
+        cls.institution = mocks.create_institution()
+        cls.institution.add_member(cls.admin)
+        cls.institution.set_admin(cls.admin.key)
+
+    def tearDown(cls):
+        """Deactivate the test."""
+        cls.test.deactivate()
+
+    
+    def test_follow(self):
+        # case in which the user is not a follower
+        self.assertEquals(self.institution.followers, [],
+            "Institution followers should be empty"
+        )
+        self.institution.follow(self.user.key)
+        self.assertEquals(self.institution.followers, [self.user.key],
+            "Institution followers should have user key"
+        )
+
+        # case in which the user is already a follower
+        self.institution.follow(self.user.key)
+        self.assertEquals(self.institution.followers, [self.user.key],
+            "Institution followers should not have the same user key more than once"
+        )
+        
+
+    def test_unfollow(self):
+        # case in which the user is not a member
+        self.institution.follow(self.user.key)
+        self.assertEquals(self.institution.followers, [self.user.key],
+            "Institution followers should have user key"
+        )
+        self.institution.unfollow(self.user.key)
```

Comment:
Shouldn't you retrieve the institution before asserting?

Reply:
I tried retrieving the institution and not retrieving it before the assertion, and the test passed anyway.